pkg/api/types.go

```diff
@@ -688,36 +691,66 @@ type ServiceStatus struct{}
 
 // ServiceSpec describes the attributes that a user creates on a service
 type ServiceSpec struct {
-	// Port is the TCP or UDP port that will be made available to each pod for connecting to the pods
-	// proxied by this service.
-	Port int `json:"port"`
+	// Required: The list of ports that are exposed.
+	// FIXME: this would be nicer as a map of name -> struct
+	Ports []ServicePort
```

Comment:
It's a new semantic and overlaps with an explicit 1-65535 range value.  It should be easy to implement - is it hypothetically useful?